src/test/java/gov/nasa/cumulus/AppTest.java

```diff
@@ -93,6 +68,17 @@ public void testError() throws IOException{
       String ex = cnm.getError(inputConfig, "WorkflowException");
       System.out.println("Exception: " + ex);
       assertNotNull(ex);
+      // Further testing with generateOutput
+      JsonObject granule = inputKey.get("input").getAsJsonObject().get("granules").getAsJsonArray().get(0).getAsJsonObject();
+      String output = CNMResponse.generateOutput(new Gson().toJson(inputConfig.get("OriginalCNM")), ex, granule, inputConfig);
+		JsonElement outputElement = new JsonParser().parse(output);
+		JsonObject response = outputElement.getAsJsonObject().get("response").getAsJsonObject();
+		JsonObject product = outputElement.getAsJsonObject().get("product").getAsJsonObject();
+		assertNotSame("SUCCESS", response.get("status").getAsString());
+		assertEquals("FAILURE", response.get("status").getAsString());
+		assertEquals("1.0", product.get("dataVersion").getAsString());
+		assertEquals(1, product.get("files").getAsJsonArray().size());
+		assertEquals("L1B_HR_SLC_product_0001-of-4154", product.get("name").getAsString());
```

Comment:
It's probably worth adding checks for uri since we have code that modifies it.

Reply:
Yes, moved these test cases to the SUCESS unit test function.